substrate/frame/examples/split/src/lib.rs

```diff
@@ -57,9 +57,7 @@ pub mod pallet {
 
 	/// Configure the pallet by specifying the parameters and types on which it depends.
 	#[pallet::config]
-	pub trait Config: frame_system::Config {
-		/// Because this pallet emits events, it depends on the runtime's definition of an event.
-		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	pub trait Config: frame_system::Config<RuntimeEvent: From<Event<Self>>> {
```

Comment:
IMO it would be much "cooler" :eyes: if the `pallet::config` macro would auto generate all of these. And then we could put a deprecation warning over `RuntimeEvent`, `RuntimeCall` etc inclusion in the `Config` trait later.

Reply:
I agree, for the implementation all pallets except frame-system can bound this `From<Event>` automatically.

For frame-system pallet, I guess we need something so that the macro can know that the pallet is `frame-system`, I suggest a new attribute somewhere (maybe on `pallet::config`) like `#[pallet::is_frame_system]`.
But if you find a better way, please do.

Reply:
ok, I have made the changes to automatically append the type bound to system supertrait. I did this on the macro expansion level instead of parsing, since we need to know if the pallet `Event` struct is actually defined.

> For frame-system pallet, I guess we need something so that the macro can know that the pallet is frame-system, I suggest a new attribute somewhere (maybe on pallet::config) like #[pallet::is_frame_system].
But if you find a better way, please do.

For this, I simply opted for checking if the pallet config contains a system supertrait, i.e `frame_system::Config`. If it doesn't, it's a `system` pallet.